tests/pipeline_backend_test.py

```diff
@@ -561,6 +573,17 @@ def test_filter(self):
             list(self.backend.filter(example_list, lambda x: x < 3)),
             [1, 2, 2, 2])
 
+    def test_filter_with_side_inputs(self):
+        data = [1, 2, 3, 4, 5, 6]
+        side_input = [[2, 4]]
+
+        def filter_fn(x, side_input):
+            return x in side_input
+
+        result = list(
+            self.backend.filter_with_side_inputs(data, filter_fn, [side_input]))
```

Comment:
`[side_input]` = `[[[2, 4]]]`, list of list of list. Is it what we want? Not list of list?

if yes, then maybe add some comment in the pipeline_backend.py because I find it confusing.

Reply:
Good point. I've added docstrings